packages/hydrogen/src/cart/optimistic/useOptimisticCart.tsx

```diff
@@ -8,23 +8,33 @@ import {
   getOptimisticLineId,
   isOptimisticLineId,
 } from './optimistic-cart.helper';
+import type {PartialDeep} from 'type-fest';
 import type {CartReturn} from '../queries/cart-types';
 
 export type OptimisticCartLine<T = CartLine> = T & {isOptimistic?: boolean};
 
-export type OptimisticCart<T = CartReturn> = Omit<T, 'lines'> & {
-  isOptimistic?: boolean;
-  lines: {
-    nodes: Array<OptimisticCartLine>;
-  };
-};
+export type OptimisticCart<T = CartReturn> = T extends {}
+  ? Omit<T, 'lines'> & {
+      isOptimistic?: boolean;
+      lines: {
+        nodes: Array<OptimisticCartLine>;
+      };
+    }
+  : // This is the null/undefined case, where the cart has yet to be created.
+    // But we still need to provide an optimistic cart object.
```

Comment:
Should we reverse this ternary op to make it more clear? `T extends null | undefined | void ? ... : ...;`?